### PR TITLE
[19570] Improve locators documentation

### PR DESCRIPTION
**Pull request by JesusPoderoso**
Fixes [#3857](https://github.com/eProsima/Fast-DDS/discussions/3857)

### CODE REVIEW

File: docs/fastdds/transport/transport_api.rst [hunk 1]
```diff
@@ -182,8 +182,11 @@ In TCP, the port of the locator is divided into a physical and a logical port.
 * The *physical port* is the port used by the network device, the real port that the operating system understands.
   It is stored in the two least significant bytes of the member |Locator_t::port-api|.
 * The *logical port* is the RTPS port.
+  It is used by the RTPS applications to perform communications.
   It is stored in the two most significant bytes of the member |Locator_t::port-api|.
 
+In TCP this distinction allows having only one opened port for external communication, while having multiple
+applications with different ports in the same host.
```
Comment (EduPonz) [inline docs/fastdds/transport/transport_api.rst:189]: I did not account for line length.
```suggestion
In TCP, this distinction allows for several DDS applications using different RTPS ports (*logical ports*) to share the same *physical port*, thus only requiring for a single port to be opened for all communications.
```

File: docs/fastdds/transport/transport_api.rst [hunk 2]
```diff
@@ -182,8 +182,11 @@ In TCP, the port of the locator is divided into a physical and a logical port.
 * The *physical port* is the port used by the network device, the real port that the operating system understands.
   It is stored in the two least significant bytes of the member |Locator_t::port-api|.
 * The *logical port* is the RTPS port.
+  It is used by the RTPS applications to perform communications.
```
Comment (EduPonz) [inline docs/fastdds/transport/transport_api.rst:185]:
```suggestion
  It is used by the RTPS protocol to distinguish different entities.
```

**Review by EduPonz — CHANGES_REQUESTED** [review]
I think we should take the opportunity also document the WAN address here

**Comment by JesusPoderoso** [comment]
> I think we should take the opportunity also document the WAN address here

I did not find the wan address in the [locators api](https://fast-dds.docs.eprosima.com/en/latest/fastdds/api_reference/rtps/common/Locator/Locator_t.html), to be referenced as done in the bullet points (`|Locator_t::port-api|` for instance)

File: docs/fastdds/transport/transport_api.rst [hunk 3]
```diff
@@ -182,14 +182,55 @@ In TCP, the port of the locator is divided into a physical and a logical port.
...
-applications with different ports in the same host.
+In TCP, this distinction allows for several DDS applications using different RTPS ports (*logical ports*) to share the
+same *physical port*, thus only requiring for a single port to be opened for all communications.
 In UDP there is only the *physical port*, which is also the RTPS port, and is stored in the two least significant bytes
 of the member |Locator_t::port-api|.
 
+The locator address, represented in 16 bytes, is managed differently depending on whether the protocol used is IPv4 or
+IPv6.
+
+* The IPv6 address uses the 16 available Bytes to represent an unique and global address.
```
Comment (EduPonz) [inline docs/fastdds/transport/transport_api.rst:196]:
```suggestion
* The IPv6 address uses the 16 available bytes to represent a unique and global address.
```

File: docs/fastdds/transport/transport_api.rst [hunk 4]
```diff
@@ -182,14 +182,55 @@ In TCP, the port of the locator is divided into a physical and a logical port.
...
+* The IPv4 address splits those 16 Bytes in the following three sections, ordered from less to greatest significant:
```
Comment (EduPonz) [inline docs/fastdds/transport/transport_api.rst:197]:
```suggestion
* The IPv4 address splits those 16 bytes in the following three sections, ordered from least to greatest significance:
```

File: docs/fastdds/transport/transport_api.rst [hunk 5]
```diff
@@ -182,14 +182,55 @@ In TCP, the port of the locator is divided into a physical and a logical port.
...
+
+  - 4 Bytes LAN IP: UDP and TCP local subnet identification.
+  - 4 Bytes WAN IP: TCP only public IP.
+  - 16 Bytes unique LAN identifier: TCP only identifier for the TCP connection.
```
Comment (EduPonz) [inline docs/fastdds/transport/transport_api.rst:201]:
```suggestion
  - 4 bytes LAN IP: local subnet identification (UDP and TCP).
  - 4 bytes WAN IP: Public IP (TCP only).
  - 8 bytes unused.
```

File: docs/fastdds/transport/transport_api.rst [hunk 6]
```diff
@@ -182,14 +182,55 @@ In TCP, the port of the locator is divided into a physical and a logical port.
...
+    |                Locator IPv6 address               | 16    |              Locator IPv4 address             | 16
+    +---------------------------------------------------+       +-----------------------------------------------+
+    | Address (2001:0000:130F:0000:0000:09C0:876A:130B) | 16    | LAN address (192.168.0.113)                   | 4
+    +---------------------------------------------------+       +-----------------------------------------------+
+                                                                | WAN address (62.128.41.210)          TCP only | 4
+                                                                +-----------------------------------------------+
+                                                                | Unique LAN ID (192.168.1.1.1.1.2.55) TCP only | 8
+                                                                +-----------------------------------------------+
+
+    TODO Edu choose which representation fits better in the documentation
```
Comment (EduPonz) [inline docs/fastdds/transport/transport_api.rst:216]: This one please, but with the header separately

File: docs/fastdds/transport/transport_api.rst [hunk 7]
```diff
@@ -182,14 +182,55 @@ In TCP, the port of the locator is divided into a physical and a logical port.
...
+
+    +--------------------------------------------------------------------------------------------------+
+    |                                        Locator IPv6 address                                      |
+    +--------------------------------------------------------------------------------------------------+
+    |                            Address (2001:0000:130F:0000:0000:09C0:876A:130B)                     |
+    +--------------------------------------------------------------------------------------------------+
+                                                   16 Bytes
+
+Check the :ref:`TCP IPv4 transport descriptor api <api_tcpv4_transport_descriptor>` for further information about WAN
+address.
```
Comment (EduPonz) [inline docs/fastdds/transport/transport_api.rst:232]: I'd rather say something in the lines of WAN address manipulation, the link is alright

File: docs/fastdds/transport/transport_api.rst [hunk 8]
```diff
@@ -182,14 +182,55 @@ In TCP, the port of the locator is divided into a physical and a logical port.
...
+                                                                | WAN address (62.128.41.210)          TCP only | 4
+                                                                +-----------------------------------------------+
+                                                                | Unique LAN ID (192.168.1.1.1.1.2.55) TCP only | 8
+                                                                +-----------------------------------------------+
+
+    TODO Edu choose which representation fits better in the documentation
+    +--------------------------------------------------------------------------------------------------+
+    |                                        Locator IPv4 address                                      |
+    +--------------------------------------+-----------------------------+-----------------------------+
+    | Unique LAN ID (192.168.1.1.1.1.2.55) | WAN address (62.128.41.210) | LAN address (192.168.0.113) |
```
Comment (EduPonz) [inline docs/fastdds/transport/transport_api.rst:220]: Please add formatting
```suggestion
    | Unused | WAN address (62.128.41.210) | LAN address (192.168.0.113) |
```

**Review by EduPonz — CHANGES_REQUESTED** [review]

**Review by EduPonz — DISMISSED** [review]
LGTM with green CI

**Comment by EduPonz** [comment]
@richiprosima please test this

**Review by EduPonz — APPROVED** [review]

**Pull request merged**